### PR TITLE
[selenium-webdriver]:added types for chromium driver

### PR DESCRIPTION
**Pull request by SinghHrmn**
- fixed missing methods from base webdriver
- chrome and edge will now extend the chromium webdriver

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/SeleniumHQ/selenium/issues/10240
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## selenium-webdriver ([<kbd>unpkg</kbd>](https://unpkg.com/browse/selenium-webdriver@latest/))
was missing the following properties:
1. withTagName
1. withLocator
1. ThenableWebDriver
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8509924226ed5420c39995641dce44c9b2224a6e
</p>

File: types/selenium-webdriver/index.d.ts [hunk 1]
```diff
@@ -2005,9 +2006,235 @@ export class WebDriver {
...
+   *     array of WebElements.
+   * @private
+   */
+  findElementsInternal_(locatorFn: Function, context: WebDriver | WebElement): Promise<WebElement[]>;
+
+  /**
+   * Creates a new WebSocket connection.
+   * @return {!Promise<resolved>} A new CDP instance.
+   */
+  createCDPConnection(target: any): Promise<any>;
```
Comment (SupernaviX) [inline types/selenium-webdriver/index.d.ts:2037]:
```suggestion
  createCDPConnection(target: string): Promise<any>;
```

File: types/selenium-webdriver/index.d.ts [hunk 2]
```diff
@@ -2005,9 +2006,235 @@ export class WebDriver {
...
+
+  /**
+   * Retrieves 'webSocketDebuggerUrl' by sending a http request using debugger address
+   * @param {string} debuggerAddress
+   * @param target
+   * @param caps
+   * @return {string} Returns parsed webSocketDebuggerUrl obtained from the http request
+   */
+  getWsUrl(debuggerAddress: string, target: any, caps: any): Promise<string>;
```
Comment (SupernaviX) [inline types/selenium-webdriver/index.d.ts:2046]:
```suggestion
  getWsUrl(debuggerAddress: string, target: string, caps: Capabilities): Promise<string>;
```

File: types/selenium-webdriver/index.d.ts [hunk 3]
```diff
@@ -2005,9 +2006,235 @@ export class WebDriver {
...
+  register(username: string, password: string, connection: any): Promise<void>;
+
+  /**
+   * Handle Network interception requests
+   * @param connection WebSocket connection to the browser
+   * @param httpResponse Object representing what we are intercepting
+   *                     as well as what should be returned.
+   * @param callback callback called when we intercept requests.
+   */
+  onIntercept(connection: WebSocket, httpResponse: any, callback: Function): Promise<void>;
```
Comment (SupernaviX) [inline types/selenium-webdriver/index.d.ts:2065]:
```suggestion
  onIntercept(connection: WebSocket, httpResponse: HttpResponse, callback: () => void): Promise<void>;
```

File: types/selenium-webdriver/index.d.ts [hunk 4]
```diff
@@ -2005,9 +2006,235 @@ export class WebDriver {
...
+
+  /**
+   *
+   * @param connection
+   * @param callback
+   * @returns {Promise<void>}
+   */
+  onLogEvent(connection: WebSocket, callback: Function): Promise<void>;
```
Comment (SupernaviX) [inline types/selenium-webdriver/index.d.ts:2073]:
```suggestion
  onLogEvent(connection: WebSocket, callback: (event: any) => void): Promise<void>;
```

File: types/selenium-webdriver/index.d.ts [hunk 5]
```diff
@@ -2005,9 +2006,235 @@ export class WebDriver {
...
+
+    /**
+   *
+   * @param connection
+   * @param callback
+   * @returns {Promise<void>}
+   */
+  onLogException(connection: WebSocket, callback: Function): Promise<void>;
```
Comment (SupernaviX) [inline types/selenium-webdriver/index.d.ts:2081]:
```suggestion
  onLogException(connection: WebSocket, callback: (event: any) => void): Promise<void>;
```

File: types/selenium-webdriver/index.d.ts [hunk 6]
```diff
@@ -2005,9 +2006,235 @@ export class WebDriver {
...
+
+  /**
+   * @param connection
+   * @param callback
+   * @returns {Promise<void>}
+   */
+  logMutationEvents(connection: WebSocket, callback: Function): Promise<void>;
```
Comment (SupernaviX) [inline types/selenium-webdriver/index.d.ts:2088]:
```suggestion
  logMutationEvents(connection: WebSocket, callback: (event: any) => void): Promise<void>;
```

File: types/selenium-webdriver/index.d.ts [hunk 7]
```diff
@@ -2005,9 +2006,235 @@ export class WebDriver {
...
+   */
+  setDownloadPath(path: string): Promise<void>;
+
+  /**
+   * Returns the list of cast sinks (Cast devices) available to the Chrome media router.
+   *
+   * @return {!promise.Thenable<void>} A promise that will be resolved with an array of Strings
+   *   containing the friendly device names of available cast sink targets.
+   */
+  getCastSinks(): Promise<void>;
```
Comment (SupernaviX) [inline types/selenium-webdriver/index.d.ts:2193]:
```suggestion
  getCastSinks(): Promise<string[]>;
```

File: types/selenium-webdriver/index.d.ts [hunk 8]
```diff
@@ -2005,9 +2006,235 @@ export class WebDriver {
...
+   *     when the mirror command has been issued to the device.
+   */
+  startCastTabMirroring(deviceName: string): Promise<void>;
+
+  /**
+   * Returns an error message when there is any issue in a Cast session.
+   * @return {!promise.Thenable<void>} A promise that will be resolved
+   *     when the mirror command has been issued to the device.
+   */
+  getCastIssueMessage(): Promise<void>;
```
Comment (SupernaviX) [inline types/selenium-webdriver/index.d.ts:2226]:
```suggestion
  getCastIssueMessage(): Promise<string>;
```

File: types/selenium-webdriver/networkinterceptor.d.ts [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+     * will be returned.
+     * @param {*} value
+     */
+    set body(arg: any);
+    get body(): any;
+    /**
+     * Sets the method of the HTTP Request
+     * @param {*} value the method of the request.
+     */
+    set method(arg: any);
```
Comment (SupernaviX) [inline types/selenium-webdriver/networkinterceptor.d.ts:38]:
```suggestion
    set method(arg: string);
```

File: types/selenium-webdriver/networkinterceptor.d.ts [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * Returns the Method to be used in the intercept
+     */
+    get method(): any;
```
Comment (SupernaviX) [inline types/selenium-webdriver/networkinterceptor.d.ts:42]:
```suggestion
    get method(): string;
```

File: types/selenium-webdriver/networkinterceptor.d.ts [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+    returnMethod: string;
+    returnStatus: number;
+    urlToIntercept: any;
+    /**
+     * Add headers that will be returned when we intercept
+     * a HTTP Request
+     * @param {*} header
+     * @param {*} value
+     */
+    addHeaders(header: any, value: any): void;
```
Comment (SupernaviX) [inline types/selenium-webdriver/networkinterceptor.d.ts:19]:
```suggestion
    addHeaders(header: string, value: any): void;
```

File: types/selenium-webdriver/networkinterceptor.d.ts [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    get headers(): any[];
```
Comment (SupernaviX) [inline types/selenium-webdriver/networkinterceptor.d.ts:20]:
```suggestion
    get headers(): { name: string, value: any }[];
```

File: types/selenium-webdriver/networkinterceptor.d.ts [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * Set the STATUS value of the returned HTTP Request
+     * @param {*} value
+     */
+    set status(arg: any);
```
Comment (SupernaviX) [inline types/selenium-webdriver/networkinterceptor.d.ts:25]:
```suggestion
    set status(arg: number);
```

File: types/selenium-webdriver/networkinterceptor.d.ts [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+    get status(): any;
```
Comment (SupernaviX) [inline types/selenium-webdriver/networkinterceptor.d.ts:26]:
```suggestion
    get status(): number;
```

File: types/selenium-webdriver/networkinterceptor.d.ts [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * Sets the value of the body of the HTTP Request that
+     * will be returned.
+     * @param {*} value
+     */
+    set body(arg: any);
```
Comment (SupernaviX) [inline types/selenium-webdriver/networkinterceptor.d.ts:32]:
```suggestion
    set body(arg: string);
```

File: types/selenium-webdriver/networkinterceptor.d.ts [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+    get body(): any;
```
Comment (SupernaviX) [inline types/selenium-webdriver/networkinterceptor.d.ts:33]:
```suggestion
    get body(): string;
```

**Review by SupernaviX — COMMENTED** [review]
Thank you for the update!

I think some of the typings are incorrect, and others could use some refinement; left some suggestions

**Review by SupernaviX — APPROVED** [review]
LGTM!

**Comment by SinghHrmn** [comment]
Ready to merge

**Pull request merged**